discovery-infra/test_infra/utils/libvirt_utils.py

```diff
@@ -0,0 +1,183 @@
+import json
+import logging
+
+import libvirt
+import xml.dom.minidom as md
+
+import waiting
+
+from logger import log
+
+from .. import consts
+
+# TODO - Temporary imports - will be removed after deleting all libvirt methods from utils.py 
+import filelock
+import os
+from contextlib import contextmanager
+
+
+# TODO - Temporary copied from utils to prevent cyclic import - 
+#  will be removed after deleting all libvirt methods from utils.py
+@contextmanager
+def _file_lock_context(filepath='/tmp/discovery-infra.lock', timeout=300):
+    logging.getLogger('filelock').setLevel(logging.ERROR)
+
+    lock = filelock.FileLock(filepath, timeout)
+    try:
+        lock.acquire()
+    except filelock.Timeout:
+        log.info(
+            'Deleting lock file: %s '
+            'since it exceeded timeout of: %d seconds',
+            filepath, timeout
+        )
+        os.unlink(filepath)
+        lock.acquire()
+
+    try:
+        yield
+    finally:
+        lock.release()
+
+
+_connection = libvirt.open("qemu:///system")
+
+
+def get_libvirt_nodes_from_tf_state(network_names, tf_state):
```

Comment:
why it is not using libvirt controller?
why we split same logic betwee controller and utils?